batch-nft-reveal/app/pages/create.tsx

```diff
@@ -17,14 +18,20 @@ function CreatePage(): JSX.Element {
   const [deployedContract, setDeployedContract] = useState<
     DeployedContract | undefined
   >()
+  const [error, setError] = useState('')
 
   const onSubmit = useCallback(
     async (args: CreateFormValues) => {
-      const tx = await deployNFTCollection(args, library?.getSigner(), chainId)
-
-      setDeployedContract({
-        address: tx.contractAddress,
-        txHash: tx.transactionHash,
+      setError('')
+      await deployNFTCollection(args, library?.getSigner(), chainId)
+      .then((tx) => {
+        setDeployedContract({
+          address: tx.contractAddress,
+          txHash: tx.transactionHash,
+        })
+      })
+      .catch((ex) => {
+        ex.message ? setError(ex.message) : setError(ex)
```

Comment:
`await` + chained `then` and `catch` is not commonly used in React, not to mention this project.